### PR TITLE
Switch to using yq version 3.3.1+

### PR DESCRIPTION
**Pull request by tomncooper**
### Type of change

Refactor

### Description

Upgrade the `yq` version to 3.3.1+ as this fixes issues with yaml indentation and other formatting issues. This will change the formatting of the generated helm files and other yaml.  

Versions of `yq` after 3.3.1 change the output of the `--version` command to go to STDERR instead of STDOUT (see [yq issue 472](https://github.com/mikefarah/yq/issues/472)). This causes the current `prerequisites-check.sh` script to fail. This PR makes so that either STDOUT or STDERR are captured. It also adds multi-platform support to the `prerequistes-check.sh`.

### CODE REVIEW

File: prerequisites-check.sh [hunk 1]
```diff
@@ -13,7 +16,12 @@ check_command_present git
 check_command_present docker
 
 YQ_VERSION="$(yq --version | ${SED} 's/^.* //g')"
+# After version 3.3.1, yq --version sends the string to STDERR instead of STDOUT
+if [ -z "$YQ_VERSION" ]; then
+    YQ_VERSION="$(yq --version 2> >(grep -Po '\d.\d.\d'))"
```
Comment (scholzj) [inline prerequisites-check.sh:21]: Thsi should probably use the GREP from the multi-platform-support.

Comment (tomncooper) [reply prerequisites-check.sh:21]: Good point.

**Comment by tomncooper** [comment]
> Do the new `yq` versions otherwise provide the same output? I.e. is the 3.3.1+ even compatible?

I have done many full image builds and tests with the version of `yq` (using `make clean all`) so I am reasonably confident it works as expected.

**Comment by scholzj** [comment]
I think doing `make crd_install` and `make helm_install` would be a good test ... that is where I saw some problems which might have been caused by this.

**Comment by tomncooper** [comment]
`make crd_install` runs fine, `helm_install` hits a new issue that apprently I now need 2 different versions of helm installed?

**Comment by tomncooper** [comment]
So I spoke too soon, `yq` 3.3.1+ seems to format the yaml for the helm charts slightly differently. By that I mean it does it correctly now, see [yq issue 343](https://github.com/mikefarah/yq/issues/343) (which was fixed in [3.3.1](https://github.com/mikefarah/yq/releases/tag/3.3.1)). 

So this means when I run `make crd_install` all the helm templates change as the indenting is different. 

What do we want to do, mandate `yq` < 3.3.1, `yq` >= 3.3.1 or just accept that whenever someone with a version either side of that line runs `make crd_install` and has to push some changes there may be a a load of whitespace diffs?

**Comment by scholzj** [comment]
Right. Just as I thought. Both ways are valid and correct. So it just sucks that they changed this :-(.

I can live with both ways ... one is friendlier to current contributors other one to new contributors. But obviously if you want to use 3.3.1+, there is lot more to change in this PR.

**Comment by tomncooper** [comment]
Do you mean I would have to add the changes in the helm charts?

**Comment by scholzj** [comment]
Well, you would need to add all the changed YAMLs and update the pipelines to use the new `yq` version as well.

**Comment by tomncooper** [comment]
The `MockConnectorTest` seems to fail a lot, but it passes locally?

**Comment by scholzj** [comment]
Why do you think that is your problem? :-o

```
ERROR: Uncommitted changes in derived resources:
M	cluster-operator/src/main/resources/cluster-roles/020-ClusterRole-strimzi-cluster-operator-role.yaml
M	cluster-operator/src/main/resources/cluster-roles/021-ClusterRole-strimzi-cluster-operator-role.yaml
M	cluster-operator/src/main/resources/cluster-roles/030-ClusterRole-strimzi-kafka-broker.yaml
M	cluster-operator/src/main/resources/cluster-roles/031-ClusterRole-strimzi-entity-operator.yaml
M	cluster-operator/src/main/resources/cluster-roles/032-ClusterRole-strimzi-topic-operator.yaml
M	install/cluster-operator/020-ClusterRole-strimzi-cluster-operator-role.yaml
M	install/cluster-operator/020-RoleBinding-strimzi-cluster-operator.yaml
M	install/cluster-operator/021-ClusterRole-strimzi-cluster-operator-role.yaml
M	install/cluster-operator/021-ClusterRoleBinding-strimzi-cluster-operator.yaml
M	install/cluster-operator/030-ClusterRole-strimzi-kafka-broker.yaml
M	install/cluster-operator/030-ClusterRoleBinding-strimzi-cluster-operator-kafka-broker-delegation.yaml
M	install/cluster-operator/031-ClusterRole-strimzi-entity-operator.yaml
M	install/cluster-operator/031-RoleBinding-strimzi-cluster-operator-entity-operator-delegation.yaml
M	install/cluster-operator/032-ClusterRole-strimzi-topic-operator.yaml
M	install/cluster-operator/032-RoleBinding-strimzi-cluster-operator-topic-operator-delegation.yaml
M	install/cluster-operator/050-Deployment-strimzi-cluster-operator.yaml
Run the following to add up-to-date resources:
  mvn clean verify -DskipTests -DskipITs \
    && make crd_install \
    && make helm_install \
    && git add install/ helm-charts/ documentation/modules/appendix_crds.adoc cluster-operator/src/main/resources/cluster-roles \
    && git commit -s -m 'Update derived resources'
```

**Comment by tomncooper** [comment]
> Why do you think that is your problem? :-o

Well there should be a `make all_the_things` target so I don't have to engage my brain 😜

**Comment by scholzj** [comment]
I'm actually not sure why the `crd_install` and `helm_install` targets are not executed at the end of `make all` since I think that would be logical. AFAIC feel free to open a PR for that.

**Comment by scholzj** [comment]
@strimzi-ci run tests profile=travis

**Comment by scholzj** [comment]
I guess you still need to update this for Jenkins.

**Comment by tomncooper** [comment]
> I'm actually not sure why the `crd_install` and `helm_install` targets are not executed at the end of `make all` since I think that would be logical. AFAIC feel free to open a PR for that.

I created #3233 for this.

**Comment by tomncooper** [comment]
@strimzi-ci run tests profile=travis

File: prerequisites-check.sh [hunk 2]
```diff
@@ -13,7 +16,22 @@ check_command_present git
...
 
 YQ_VERSION="$(yq --version | ${SED} 's/^.* //g')"
-if [[ ${YQ_VERSION} != "3."* ]]; then
-  echo -e "${RED}yq vesion is ${YQ_VERSION}, version must be 3.*${NO_COLOUR}"
+# After version 3.3.1, yq --version sends the string to STDERR instead of STDOUT
+if [ -z "$YQ_VERSION" ]; then
+    YQ_VERSION="$(yq --version 2> >($GREP -Po '\d.\d.\d'))"
+fi
+
+yq_array=($(echo "$YQ_VERSION" | tr '.' '\n'))
```
Comment (scholzj) [inline prerequisites-check.sh:24]: Do we still need to be able to handle both 3.3.1 as well as the previous versions on line 18? Or could we just remove it?

Comment (tomncooper) [reply prerequisites-check.sh:24]: The old version sends the version string to STDOUT so we need that line to capture that and tell the user to upgrade. The new version (3.3.1+) sends it to STDERR so we need to cover both to reliably check the version of `yq`.

**Comment by scholzj** [comment]
@strimzi/maintainers Can we get a second review for this?

**Comment by ppatierno** [comment]
Are all these changes on YAMLs related to the new yq version?

**Comment by tomncooper** [comment]
> Are all these changes on YAMLs related to the new yq version?

Yes. `yq` 3.3.1+ uses the correct YAML array indentation level (the old one did not indent list elements compared to the list level). That means any lists in the generated YAML will show whitespace diffs.

**Pull request merged**